### ISSUE TITLE
`withenv` only accepts `Function`s

### ISSUE THREAD

**Issue by ericphanson** [JuliaLang/julia#43486]
https://github.com/JuliaLang/julia/blob/bb5b98e72a151c41471d8cc14cacb495d647fb7f/base/env.jl#L157-L179

This means you can't use it with callable types. Is this restriction needed? Seems like we could just drop it.

**Comment by DilumAluthge** [comment]
As a workaround, could you do e.g. `withenv(() -> x(), ...)`, where `x` is the callable object?

**Comment by ericphanson** [comment]
Yeah, I'm doing that in my code, but I figured it could be fixed here too.
